### PR TITLE
Fjm create post

### PR DESCRIPTION
**Pull request by fmings**
files and functions to create post
currently publicationDate and userId are set to static values and need to be updated to host actual date/time
**Not sure userId can be set to dynamic value since there is no login

### CODE REVIEW

**Review by FletcherJMoore — APPROVED** [review]

File: components/forms/PostForm.js [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    setFormInput((prevState) => ({
+      ...prevState,
+      [name]: value,
+    }));
+  };
+
+  const handleSubmit = (p) => {
+    p.preventDefault();
+    // ADDED PAYLOAD VARIABLE, ADDED USERID AND PUBLICATION DATE KEYS - NEED TO FETCH USER ID AND REPLACE THE TEST CONSTANT BELOW; NEED TO ALSO ENSURE PUBLICATIONDATE IS PULLING IN CORRECT DATE/TIME DATA, CURRENT VALUE IS A DUMMY FOR TESTING
+    const payload = { ...formInput, userId: 1, publicationDate: Date.now };
```
Comment (britnay268) [inline components/forms/PostForm.js]: Instead of Date.now - use new Date() and remove the userId

File: components/forms/PostForm.js [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+      updatePost(payload).then(() => router.push('/posts'));
+    } else {
+      createPost(payload).then(() => router.push('/posts'));
+    }
+  };
+
+  return (
+    <Form onSubmit={handleSubmit}>
+      <h2 className="text-white mt-5">{obj.id ? 'Update' : 'Create'} Post</h2>
+
```
Comment (britnay268) [inline components/forms/PostForm.js:59]: We can insert a user dropdown to assign the right userId/user to the post.      
<FloatingLabel controlId="floatingSelect" label="User">
        <Form.Select
          aria-label="User"
          name="userId"
          onChange={handleChange}
          className="mb-3"
          // CHANGED KEY TO FORMINPUT VERSUS OBJ
          value={formInput.userId}
          required
        >
          <option value="">Select the user creating the Post</option>
          {
          users.map((u) => (
            <option
              key={u.id}
              value={u.id}
            >
              {u.firstName} {u.lastName}
            </option>
          ))
        }
        </Form.Select>
      </FloatingLabel>

File: components/forms/PostForm.js [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+  content: '',
+  approved: false,
+};
+
+function PostForm({ obj }) {
+  const [formInput, setFormInput] = useState(initialState);
+  // const [tags, setTags] = useState([]);
+  const [categories, setCategories] = useState([]);
+  const router = useRouter();
+
```
Comment (britnay268) [inline components/forms/PostForm.js:29]: Probably create a state for users to store all users like what was done for categories.

**Review by britnay268 — CHANGES_REQUESTED** [review]

**Review by Zpcolburn — APPROVED** [review]
Works just as it needs to.

**Review by britnay268 — APPROVED** [review]

**Pull request merged**